File: src/types/game.ts
import type { ComponentProps } from 'react';
import type { Ionicons } from '@expo/vector-icons';

export type IoniconName = ComponentProps<typeof Ionicons>['name'];

export type CaseCategoryId =
  | 'people'
  | 'characters'
  | 'animals'
  | 'places'
  | 'objects'
  | 'games'
  | 'brands'
  | 'anything';

export type InvestigationMode = 'standard' | 'rapid';

export type AnswerValue = 'yes' | 'no' | 'maybe' | 'unknown';

export type GameStatus =
  | 'idle'
  | 'playing'
  | 'thinking'
  | 'guessing'
  | 'won'
  | 'lost';

export interface CategoryOption {
  id: CaseCategoryId;
  label: string;
  icon: IoniconName;
  description: string;
  emphasized?: boolean;
}

export interface ModeOption {
  id: InvestigationMode;
  title: string;
  subtitle: string;
  questionCount: number;
}

export interface QAEntry {
  question: string;
  answer: AnswerValue;
}

export interface CandidateGuess {
  name: string;
  confidence: number;
  reason?: string;
}

export interface CategoryBreakdownItem {
  label: string;
  percentage: number;
}

export interface InvestigationSnapshot {
  categoryBreakdown: CategoryBreakdownItem[];
  candidatesRemaining: number;
  aiConfidence: number;
  topPossibilities: string[];
}

export interface GameState {
  gameId: string;
  caseNumber: number;
  category: CaseCategoryId | null;
  mode: InvestigationMode;
  questionNumber: number;
  maxQuestions: number;
  questions: string[];
  answers: QAEntry[];
  candidates: string[];
  confidence: number;
  status: GameStatus;
  currentQuestion: string;
  guess: CandidateGuess | null;
  snapshot: InvestigationSnapshot;
}

export type CaseResultType = 'ai_victory' | 'player_victory';

export interface CaseRecord {
  id: string;
  caseNumber: number;
  subject: string;
  category: CaseCategoryId;
  questionsUsed: number;
  maxQuestions: number;
  result: CaseResultType;
  confidence: number;
  score: number;
  date: string;
}

File: backend/app/schemas/game.py
from pydantic import BaseModel, Field
from typing import List, Optional
from uuid import UUID

class GameState(BaseModel):
    game_id: UUID
    category: str
    mode: str
    difficulty: str = "normal"
    personality: str = "analytical"
    question_number: int = 0
    max_questions: int
    questions: List[str] = Field(default_factory=list)
    answers: List[str] = Field(default_factory=list)
    status: str = "idle"  # idle, playing, thinking, guessing, won, lost
    confidence: int = 0
    candidates: List[str] = Field(default_factory=list)
    current_question: Optional[str] = None
    guess: Optional[str] = None
    reason: Optional[str] = None
    target_entity: Optional[str] = None
    contradiction: Optional[str] = None
    is_daily: bool = False
    has_lied: bool = False
    lie_index: int = -1
    lie_caught: bool = False
    fact_checks: int = 1
    desperation_clue: Optional[str] = None
    pending_desperation_clue: Optional[str] = None
    transcript: Optional[str] = None

class CreateGameRequest(BaseModel):
    category: str
    mode: str
    difficulty: str = "normal"
    personality: str = "analytical"

class AnswerRequest(BaseModel):
    answer: str

class ConfirmGuessRequest(BaseModel):
    correct: bool

class DesperationRequest(BaseModel):
    clue: str
